app/src/main/java/com/xmartlabs/scasas/doapp/ui/GroupsListFragment.java

```diff
@@ -0,0 +1,471 @@
+            closeNewTaskView();
+          }
+
+          @Override
+          public void onError(Throwable error) {
+            Timber.e(error);
+          }
+        });
+  }
+
+  private void updateItemCount(String groupName) {
+    int group = getStringRes(groupName);
+    if (group != 0) {
+      updateItemCount(group);
+    }
+  }
+
+  private void updateItemCount(@StringRes int groupName) {
+    switch (groupName) {
+      case R.string.shop:
+        getTaskCount(R.string.shop);
+        break;
+      case R.string.work:
+        getTaskCount(R.string.work);
+        break;
+      case R.string.health:
+        getTaskCount(R.string.health);
+        break;
+      case R.string.travel:
+        getTaskCount(R.string.travel);
+        break;
+      case R.string.bills:
+        getTaskCount(R.string.bills);
+        break;
+      case R.string.auto:
+        getTaskCount(R.string.auto);
+        break;
+      default:
+        updateAll(groups);
+    }
+  }
+
+  private void updateAll(List<Group> groups) {
+    Stream.of(groups).forEach(this::getTaskCount);
+  }
+
+  private void getTaskCount(@StringRes int groupName) {
+    Group group = getGroup(groupName);
+    taskController.getTasksCount(user, group).subscribe(new SingleSubscriber<Long>() {
+      @Override
+      public void onSuccess(Long taskCount) {
+        int groupName = getStringRes(group.getName());
+        updateItemField(groupName, taskCount);
+      }
+
+      @Override
+      public void onError(Throwable error) {
+        Timber.e(error);
+      }
+    });
+  }
+
+  private void getTaskCount(Group group) {
+    taskController.getTasksCount(user, group).subscribe(new SingleSubscriber<Long>() {
+      @Override
+      public void onSuccess(Long taskCount) {
+        int groupName = getStringRes(group.getName());
+        updateItemField(groupName, taskCount);
+      }
+
+      @Override
+      public void onError(Throwable error) {
+        Timber.e(error);
+      }
+    });
+  }
+
+  private String getTextFromStringRes(@StringRes int text) {
+    return getResources().getString(text);
+  }
+
+  private int getStringRes(String text) {
+    if (text.equals(getTextFromStringRes(R.string.shop))) {
+      return R.string.shop;
+    }
+    if (text.equals(getTextFromStringRes(R.string.work))) {
+      return R.string.work;
+    }
+    if (text.equals(getTextFromStringRes(R.string.health))) {
+      return R.string.health;
+    }
+    if (text.equals(getTextFromStringRes(R.string.travel))) {
+      return R.string.travel;
+    }
+    if (text.equals(getTextFromStringRes(R.string.bills))) {
+      return R.string.bills;
+    }
+    if (text.equals(getTextFromStringRes(R.string.auto))) {
+      return R.string.auto;
+    }
+    return 0;
+  }
+
+  private void updateItemField(@StringRes int groupName, long itemCount) {
+    switch (groupName) {
+      case R.string.shop:
+        shopItemsView.setText(String.format(getString(R.string.item), itemCount));
```

Comment:
I think that using a View variable, and asign it in the switch is more clear. 
```
View itemView; 
switch (groupName) {
       case R.string.shop:
         shopItemsView
.....
itemView.setText(String.format(getString(R.string.item), itemCount));
```
However, it's only a comment